skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,148 @@
+    Unsharp masking is an image sharpening technique. It is a linear image
+    operation, and numerically stable, unlike deconvolution which is an
+    ill-posed problem. Because of this stability, it is often more
+    preferred than deconvolution.
+
+    The main idea is the following: Sharp details are identified as the
+    difference between the original image and its blurred version.
+    These details are added back to the original image after a scaling step:
+
+        enhanced image = original + amount * (original - blurred)
+
+    Despite being relatively simple algorithm, it might have issues when
+    RGB color channels are processed, i.e., it might lead to color bleading.
+    Often visually more pleasing result can be achieved with processing
+    only the brightness/lightness/intensity channel in a suitable color space,
+    like HSV, HSL, YUV, YCbCr, etc. In these case, a color space conversion step
+    should be performed before and after the unsharp masking.
+    Note that some color spaces have different than [0,1] / [0,255] dynamic range,
+    and it is the user's responsibility to take this into account.
+    See the skimage.color module for details about color space conversion.
+
+    Unsharp mask filter is described in most of the digital image processing
+    books. This implementation is based on [1]_.
+
+    Examples
+    --------
+    >>> array = np.ones(shape=(5,5), dtype=np.uint8)*100
+    >>> array[2,2] = 120
+    >>> array
+    array([[100, 100, 100, 100, 100],
+           [100, 100, 100, 100, 100],
+           [100, 100, 120, 100, 100],
+           [100, 100, 100, 100, 100],
+           [100, 100, 100, 100, 100]], dtype=uint8)
+    >>> np.around(unsharp_mask(array, radius=0.5, amount=2),2)
+    array([[ 0.39,  0.39,  0.39,  0.39,  0.39],
+           [ 0.39,  0.39,  0.39,  0.39,  0.39],
+           [ 0.39,  0.39,  0.47,  0.39,  0.39],
+           [ 0.39,  0.39,  0.39,  0.39,  0.39],
+           [ 0.39,  0.39,  0.39,  0.39,  0.39]])
+
+    >>> array = np.ones(shape=(5,5), dtype=np.int8)*100
+    >>> array[2,2] = 127
+    >>> np.around(unsharp_mask(array, radius=0.5, amount=2),2)
+    array([[ 0.79,  0.79,  0.79,  0.79,  0.79],
+           [ 0.79,  0.79,  0.79,  0.79,  0.79],
+           [ 0.79,  0.79,  1.  ,  0.79,  0.79],
+           [ 0.79,  0.79,  0.79,  0.79,  0.79],
+           [ 0.79,  0.79,  0.79,  0.79,  0.79]])
+
+    >>> np.around(unsharp_mask(array, radius=0.5, amount=2, preserve_range=True),2)
+    array([[ 100.  ,  100.  ,   99.99,  100.  ,  100.  ],
+           [ 100.  ,   99.39,   95.48,   99.39,  100.  ],
+           [  99.99,   95.48,  147.59,   95.48,   99.99],
+           [ 100.  ,   99.39,   95.48,   99.39,  100.  ],
+           [ 100.  ,  100.  ,   99.99,  100.  ,  100.  ]])
+
+
+    References
+    ----------
+    .. [1]  Jayaraman, S., Esakkirajan, S., Veerakima, T.
+            "Digital image processing", (2009), ed iii., page 274,
+            ISBN 10: 0070144796  ISBN 13: 9780070144798
```

Comment:
Add DOI, please.

Reply:
I cannot. It is a book, it has no DOI, but ISBN.
I rather give a different book as a reference, this one has a DOI.

Reply:
I mean, some books have DOIs, but this one did not have.

Reply:
Yes, sorry, my bad. Let's keep the current reference.